src/main/java/de/bonndan/nivio/output/map/svg/SVGRenderer.java

```diff
@@ -1,33 +1,113 @@
+import de.bonndan.nivio.output.map.hex.Hex;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 import org.springframework.stereotype.Service;
+import org.springframework.util.StringUtils;
 
+import java.awt.geom.Point2D;
 import java.io.File;
 import java.io.FileWriter;
 import java.io.IOException;
+import java.util.concurrent.atomic.AtomicLong;
 
 @Service
 public class SVGRenderer implements Renderer<String> {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(SVGRenderer.class);
     public static final int DEFAULT_ICON_SIZE = 50;
+
+    private final LocalServer localServer;
     private final MapStyleSheetFactory mapStyleSheetFactory;
 
-    public SVGRenderer(MapStyleSheetFactory mapStyleSheetFactory) {
+    public SVGRenderer(LocalServer localServer, MapStyleSheetFactory mapStyleSheetFactory) {
+        this.localServer = localServer;
         this.mapStyleSheetFactory = mapStyleSheetFactory;
     }
 
     @Override
-    public String render(LandscapeImpl landscape) {
-        SVGDocument svgDocument = new SVGDocument(landscape, mapStyleSheetFactory);
+    public String render(LayoutedComponent layoutedComponent) {
+        applyValues(layoutedComponent);
+        SVGDocument svgDocument = new SVGDocument(layoutedComponent, mapStyleSheetFactory);
         return svgDocument.getXML();
     }
 
     @Override
-    public void render(LandscapeImpl landscape, File file) throws IOException {
+    public void render(LayoutedComponent landscape, File file) throws IOException {
         FileWriter fileWriter = new FileWriter(file);
         fileWriter.write(render(landscape));
         fileWriter.close();
     }
+
+    /**
+     * @param layoutedComponent layouted landscape
+     */
+    private void applyValues(LayoutedComponent layoutedComponent) {
+
+        Point2D.Double margin = getMargins(layoutedComponent);
+        layoutedComponent.getChildren().forEach(groupBounds -> {
+
+            LOGGER.debug("group offset {} {}", groupBounds.getX(), groupBounds.getY());
+            Group group = (Group) groupBounds.getComponent();
+            if (StringUtils.isEmpty(group.getColor())) {
+                group.setColor(Color.getGroupColor(group));
+            }
+            groupBounds.setX(groupBounds.getX() + margin.x);
+            groupBounds.setY(groupBounds.getY() + margin.y);
+
+            groupBounds.getChildren().forEach(itemBounds -> {
+                LOGGER.debug("original item pos {} {}", itemBounds.getX(), itemBounds.getY());
+                itemBounds.setX(itemBounds.getX() + groupBounds.getX());
+                itemBounds.setY(itemBounds.getY() + groupBounds.getY());
+                LOGGER.debug("item pos with group offset: {} {}", itemBounds.getX(), itemBounds.getY());
+
+                Item item = (Item) itemBounds.getComponent();
+                itemBounds.setColor(group.getColor());
+                itemBounds.setIcon(localServer.getIconUrl(item).toString());
```

Comment:
This will link the image from 'http://localhost:8080/icons/service.png'
maybe it is better to embed the image base64 encoded.
I think this is also the reason why 'https://github.com/dedica-team/nivio/actions/runs/183402402' fails.

Reply:
fixed